File: backend/app/models/experience.py
import uuid
from sqlalchemy import Column, String, Date, ForeignKey
from app.database import Base, TimestampMixin


class Experience(Base, TimestampMixin):
    __tablename__ = "experiences"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()), index=True)
    user_id = Column(String(36), ForeignKey("users.id"), nullable=False, index=True)

    organisation_name = Column(String, nullable=False)
    role = Column(String, nullable=False)
    start_date = Column(Date, nullable=True)
    end_date = Column(Date, nullable=True)        # null = currently working (per contract)
    description = Column(String, nullable=True)
    type = Column(String, nullable=False)          # Internship / Part-time / Full-time

File: frontend/src/api/entities/experience.ts
/**
 * Work experience entity.
 * Represents an internship, part-time, or full-time professional role.
 */

export type ExperienceType = 'Internship' | 'Part-time' | 'Full-time';

export interface Experience {
  id: string;
  organisationName: string;
  role: string;
  startDate: string;
  endDate?: string;
  description: string;
  type: ExperienceType;
  createdAt: string;
  updatedAt: string;
}
